pkg/kubelet/secret/secret_manager.go

```diff
@@ -145,6 +145,12 @@ func (s *secretStore) Delete(namespace, name string) {
 	}
 }
 
+// fromApiserverCache modifies <opts> so that the GET request will
```

Comment:
How about moving this to the util package? This can be used for [getting the node object](https://github.com/kubernetes/kubernetes/blob/a507777de86f9a64ac0a1ff1ed98f7e1e23f8f4d/pkg/kubelet/kubelet_node_status.go#L352) too :-)

Other than that the rest of the PR LGTM